src/DocNet/NavigationLevel.cs

```diff
@@ -31,198 +31,325 @@
-				}
-				fragments.Add("</ul>");
-			}
-			else
-			{
-				// just a link
-				fragments.Add(string.Format("<span class=\"navigationgroup\"><i class=\"fa fa-caret-right\"></i> <a href=\"{0}{1}\">{2}</a></span>", 
-											relativePathToRoot, HttpUtility.UrlPathEncode(this.TargetURL), this.Name));
-			}
-			if(!this.IsRoot)
-			{
-				fragments.Add("</li>");
-			}
-			return string.Join(Environment.NewLine, fragments.ToArray());
-		}
-
-
-		#region Properties
-		public override string TargetURL
-		{
-			get
-			{
-				var defaultElement = this.IndexElement;
-				if(defaultElement == null)
-				{
-					return string.Empty;
-				}
-				return defaultElement.TargetURL ?? string.Empty;
-			}
-		}
-
-
-		public SimpleNavigationElement IndexElement
-		{
-			get
-			{
-				var toReturn = this.Value.FirstOrDefault(e => e.IsIndexElement) as SimpleNavigationElement;
-				if(toReturn == null)
-				{
-					// no index element, add an artificial one.
-					var path = string.Empty;
-					if(this.ParentContainer != null)
-					{
-						path = Path.GetDirectoryName(this.ParentContainer.TargetURL);
-					}
-					var nameToUse = this.Name.Replace(".", "").Replace('/', '_').Replace("\\", "_").Replace(":", "").Replace(" ", "");
-					if(string.IsNullOrWhiteSpace(nameToUse))
-					{
-						return null;
-					}
-					toReturn = new SimpleNavigationElement() {ParentContainer = this, Value = string.Format("{0}{1}.md", path, nameToUse), Name = this.Name, IsIndexElement = true};
-					this.Value.Add(toReturn);
-				}
-
-				return toReturn;
-			}
-		}
-
-
-		/// <summary>
-		/// Gets / sets a value indicating whether this element is the __index element
-		/// </summary>
-		public override bool IsIndexElement
-		{
-			// never an index
-			get { return false; }
-			set {  
-				// nop;
-			}
-		}
-
-
-		public bool IsRoot { get; set; }
-		#endregion
-	}
+    public class NavigationLevel : NavigationElement<List<INavigationElement>>
+    {
+        private readonly string _rootDirectory;
+
+        public NavigationLevel(string rootDirectory)
+            : base()
+        {
+            this._rootDirectory = rootDirectory;
+            this.Value = new List<INavigationElement>();
+        }
+
+
+        public void Load(JObject dataFromFile)
+        {
+            foreach (KeyValuePair<string, JToken> child in dataFromFile)
+            {
+                INavigationElement toAdd;
+                if (child.Value.Type == JTokenType.String)
+                {
+                    var nameToUse = child.Key;
+
+                    var isIndexElement = child.Key == "__index";
+                    if (isIndexElement)
+                    {
+                        nameToUse = this.Name;
+                    }
+
+                    var childValue = child.Value.ToObject<string>();
+
+                    var endsWithWildcards = childValue.EndsWith("**");
+                    if (endsWithWildcards)
```

Comment:
`endsWithWildcards` is used just once, you can inline the variable in the if, which is preferable.